blivet/devicefactory.py

```diff
@@ -294,50 +301,70 @@ def __init__(self, storage, size, disks, fstype=None, mountpoint=None,
             :keyword container_encrypted: whether to encrypt the container
             :type container_encrypted: bool
             :keyword container_size: requested container size
-            :type container_size: :class:`~.size.Size`
+            :type container_size: :class:`~.size.Size`, :const:`SIZE_POLICY_AUTO` (the default),
+                                  or :const:`SIZE_POLICY_MAX`
             :keyword min_luks_entropy: minimum entropy in bits required for
                                        LUKS format creation
             :type min_luks_entropy: int
 
         """
-        self.storage = storage          # a Blivet instance
-        self.size = size                # the requested size for this device
-        self.disks = disks              # the set of disks to allocate from
-
-        self.original_size = size
-        self.original_disks = disks[:]
-
-        self.fstype = fstype
-        self.mountpoint = mountpoint
-        self.label = label
-
-        self.raid_level = raid_level
-        self.container_raid_level = container_raid_level
-
-        self.encrypted = encrypted
-        self.container_encrypted = container_encrypted
-
-        self.container_name = container_name
-        self.device_name = name
-
-        self.container_size = container_size
-
-        self.container = None
-        self.device = device
+        self.storage = kwargs["storage"]  # a Blivet instance
+
+        self._encrypted = None
+        self._raid_level = None
+        self._container_raid_level = None
+
+        for setting, value in self._default_settings.items():
+            setattr(self, setting, value)
+
+        self.fstype = None  # not included in default_settings b/c of special handling below
```

Comment:
I left it out of the dict because the below block prevents unit testing based on the dict from passing.